I built all the supported ocamls, and it seems the fairly old 4.9 (March 2020) and 4.10 (December 2020) compilers cannot even run the tests as they get the following error:

```
ocaml> File "testing.ml", line 1:
ocaml> Error: Could not find the .cmi file for interface testing.mli.
ocaml> File "lexcmm.mli", line 16, characters 28-42:
ocaml> 16 | val token: Lexing.lexbuf -> Parsecmm.token
ocaml>                                  ^^^^^^^^^^^^^^
ocaml> Error: Unbound module Parsecmmmake[3]: *** [Makefile:46: testing.cmx] Error 2
```

Update the PR to disable checks on these two older versions.

--- a/pkgs/development/compilers/ocaml/4.09.nix
+++ b/pkgs/development/compilers/ocaml/4.09.nix
@@ -7,6 +7,9 @@ import ./generic.nix {
   # Breaks build with Clang
   hardeningDisable = [ "strictoverflow" ];
 
+  # Tests do not seem to run on this old version
+  doCheck = false;
+
   patches = [
     ./4.09.1-Werror.patch
     # Compatibility with Glibc 2.34

--- a/pkgs/development/compilers/ocaml/4.10.nix
+++ b/pkgs/development/compilers/ocaml/4.10.nix
@@ -3,6 +3,10 @@ import ./generic.nix {
   minor_version = "10";
   patch_version = "2";
   sha256 = "sha256-locUYQeCgtXbAiB32JveJchfteN2YStE+MN9ToTwAOM=";
+
+  # Tests do not seem to run on this old version
+  doCheck = false;
+
   patches = [
     ./glibc-2.34-for-ocaml-4.10-and-11.patch
   ];

--- a/pkgs/development/compilers/ocaml/generic.nix
+++ b/pkgs/development/compilers/ocaml/generic.nix
@@ -3,6 +3,7 @@
   major_version,
   patch_version,
   patches ? [ ],
+  doCheck ? true,
   ...
 }@args:
 let
@@ -83,6 +84,7 @@ let
 in
 
 stdenv.mkDerivation (
+  finalArgs:
   args
   // {
 
@@ -115,7 +117,9 @@ stdenv.mkDerivation (
         "-host ${stdenv.hostPlatform.config}"
         "-target ${stdenv.targetPlatform.config}"
       ]
-      ++ optional noNakedPointers (flags "--disable-naked-pointers" "-no-naked-pointers");
+      ++ optional noNakedPointers (flags "--disable-naked-pointers" "-no-naked-pointers")
+      ++ optional finalArgs.doCheck "--enable-ocamltest";
+
     dontAddStaticConfigureFlags = lib.versionOlder version "4.08";
 
     env =
@@ -194,6 +198,9 @@ stdenv.mkDerivation (
     passthru = {
       nativeCompilers = useNativeCompilers;
     };
+
+    checkTarget = "tests";
+    inherit doCheck;
 
     meta = {
       homepage = "https://ocaml.org/";

--- a/pkgs/development/compilers/ocaml/generic.nix
+++ b/pkgs/development/compilers/ocaml/generic.nix
@@ -204,6 +204,8 @@ stdenv.mkDerivation (
       ];
       description = "OCaml is an industrial-strength programming language supporting functional, imperative and object-oriented styles";
 
+      maintainers = [ lib.maintainers.georgyo ];
+
       longDescription = ''
         OCaml is a general purpose programming language with an emphasis on expressiveness and safety. Developed for more than 20 years at Inria by a group of leading researchers, it has an advanced type system that helps catch your mistakes without getting in your way. It's used in environments where a single mistake can cost millions and speed matters, is supported by an active community, and has a rich set of libraries and development tools. It's widely used in teaching for its power and simplicity.
 

